package.json

```diff
@@ -34,7 +36,12 @@
     "babel-eslint": "^7.2.2",
     "eslint": "^4.17.0",
     "eslint-config-synacor": "^3.0.0",
-    "mkdirp": "^0.5.1"
+    "mkdirp": "^0.5.1",
+    "mockery": "^2.1.0",
+    "preact": "^8.2.9",
```

Comment:
I'm not sure these are dependencies, they are external to `zimlet-cli` so I think we just leave them out and let them be dependencies in `zm-x-web`.

Reply:
Do you think this is worth doing or should we just update it by hand?

Reply:
This seems much easier than keeping track of every named export for every module we want to shim.

Do you think it's worth adding a runtime check for undefined exports and warn consumers?

```js
`
${Object.keys(shimModule)}.forEach((method) => {
  if (!(method in global.shim.foo)) {
    console.warn(\`[ZimletCLI]: Export foo[\${method}] was not found in global environment.\`);
  }
});
`
```

Reply:
That's really only a problem if a zimlet is going to use one of those undefined methods, in which case an exception would be thrown when it tried to use it.  It's balancing spamming the console with helping to debug...

Maybe an alternative would be to do something like this to only warn on usage:
```js
${Object.keys(shimModule)}.forEach((method) => {
  if (!(method in global.shim.foo)) {
    global.shim.foo = () => {
         throw new Error(`[ZimletCLI]: Export foo[${method}] was not found in global environment.`);
    }
   
  }
});
```

Reply:
I like that, I think it's the best balance.

That gives the developer some breadcrumbs to follow without spamming; otherwise a developer unaware of the shim setup would be horribly confused.

Reply:
My latest commit added this wrapper function around each exported name.  I tried to be conservative in how I did the exports so that webpack tree-shaking wouldn't get broken.